packs/aws/actions/lib/action.py

```diff
@@ -45,16 +45,25 @@ def split_tags(self, tags):
                 tag_dict[k] = v
         return tag_dict
 
-    def wait_for_state(self, instance_id, state):
+    def wait_for_state(self, instance_id, state, timeout=10, retries=3):
         state_list = {}
         obj = self.ec2_connect()
-        time.sleep(10)
-        for instance in obj.get_only_instances([instance_id, ]):
+        eventlet.sleep(timeout)
+        for _ in range(retries + 1):
+            try:
+                instance_list = obj.get_only_instances([instance_id, ])
+            except Exception:
+                self.logger.info("Waiting for instance to become available")
+                eventlet.sleep(timeout)
+
+        for instance in instance_list:
```

Comment:
instance_list could be undefined at this point if all attempts throw an exception. Assign it a default value "[ ]" outside the for loop